docs/contracts/constant-state-variables.rst

```diff
@@ -86,14 +103,3 @@ you are comparing the
 runtime code generated by the compiler with the one actually stored in the
 blockchain. The compiler outputs where these immutables are located in the deployed bytecode
 in the ``immutableReferences`` field of the :ref:`compiler JSON standard output <compiler-api>`.
-
-.. note::
-  Immutables that are assigned at their declaration are only considered
-  initialized once the constructor of the contract is executing.
-  This means you cannot initialize immutables inline with a value
-  that depends on another immutable. You can do this, however,
-  inside the constructor of the contract.
```

Comment:
We actually did. We used to keep a list of already initialized immutables in `m_initializedStateVariables` and it was only filled just before visiting the corresponding constructor (rather than being updated immediately after visiting each variable).

As a result you were not able to refer to other immutables in initializers (regardless of declaration order):
```solidity
contract C {
    uint immutable x = y; // Error: Immutable variables cannot be read before they are initialized.
    uint immutable y = 1;
}
```
```solidity
contract C {
    uint immutable x = 1;
    uint immutable y = x; // Error: Immutable variables cannot be read before they are initialized.
}
```

Reply:
Good enough for now.